File: src/main/java/com/divide2/dic/controller/LabelController.java
package com.divide2.dic.controller;

import com.divide2.constant.Responser;
import com.divide2.constant.ReturnCoder;
import com.divide2.dic.model.Label;
import com.divide2.dic.service.LabelService;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Resource;
import java.util.List;

/**
 * create by bvvy
 */
@Controller
@RequestMapping("/v1/label")
public class LabelController {

    @Resource
    private LabelService labelServiceImpl;

    @GetMapping("/all")
    public ResponseEntity<List<Label>> all() {
        return Responser.ok(labelServiceImpl.all());
    }

    @GetMapping("{id}")
    public ResponseEntity<Label> get(@PathVariable Integer id) {
        return Responser.ok(labelServiceImpl.get(id));
    }

    @DeleteMapping("{id}")
    public ResponseEntity<ReturnCoder> delete(@PathVariable Integer id) {
        labelServiceImpl.delete(id);
        return Responser.delete();
    }

    @PostMapping
    public ResponseEntity<ReturnCoder> add(@RequestBody Label label) {
        labelServiceImpl.add(label);
        return Responser.created();
    }
    @PatchMapping
    public ResponseEntity<ReturnCoder> update(@RequestBody Label label) {
        labelServiceImpl.update(label);
        return Responser.update();
    }
    
}

File: src/main/kotlin/com/divide2/module/label/LabelController.kt
package com.divide2.module.label

import com.divide2.er.COMMA
import com.divide2.er.Messager
import com.divide2.er.Responser
import org.springframework.data.domain.Pageable
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

/**
 * create by bvvy
 */

@RestController
@RequestMapping("/v1/label")
class LabelController(val labelRepository: LabelRepository) {

    @GetMapping("/all")
    fun all() = Responser.ok(labelRepository.findAll())

    @GetMapping("/{id}")
    fun get(@PathVariable id: Int) = Responser.ok(labelRepository.getOne(id))

    @DeleteMapping("/{ids}")
    fun delete(@PathVariable ids: String): ResponseEntity<Messager> {
        ids.split(COMMA).forEach { labelRepository.deleteById(it.toInt()) }
        return Responser.delete()
    }

    @PostMapping
    fun add(@RequestBody label: Label): ResponseEntity<Messager> {
        labelRepository.save(label)
        return Responser.created()
    }

    @PatchMapping
    fun update(@RequestBody label: Label): ResponseEntity<Messager> {
        labelRepository.save(label)
        return Responser.update()
    }

    @GetMapping("/list")
    fun page(pageable: Pageable) = labelRepository.findAll(pageable)



}

